File: app/src/main/java/com/openclassrooms/realestatemanager/models/Media.java
package com.openclassrooms.realestatemanager.models;

import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Index;
import androidx.room.PrimaryKey;

@Entity(indices = {@Index("propertyId")},
        foreignKeys = @ForeignKey(entity = Property.class,
                parentColumns = {"id"},
                childColumns = {"propertyId"})
        )
public class Media {

    @PrimaryKey(autoGenerate = true)
    private long id;
    private String label;
    private String fileName;
    private boolean isCover;

    private long propertyId; // FK

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getLabel() {
        return label;
    }

    public void setLabel(String label) {
        this.label = label;
    }

    public String getFileName() {
        return fileName;
    }

    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    public long getPropertyId() {
        return propertyId;
    }

    public void setPropertyId(long propertyId) {
        this.propertyId = propertyId;
    }

    public boolean isCover() {
        return isCover;
    }

    public void setCover(boolean cover) {
        isCover = cover;
    }
}

File: app/src/main/java/com/openclassrooms/realestatemanager/models/Media.kt
package com.openclassrooms.realestatemanager.models

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(indices = [Index("propertyId")],
        foreignKeys = [ForeignKey(entity = Property::class,
                parentColumns = ["id"],
                childColumns = ["propertyId"])])
class Media {

    @PrimaryKey(autoGenerate = true)
    var id: Long = 0
    var label: String? = null
    var fileName: String? = null
    var isCover: Boolean = false

    var propertyId: Long = 0 // FK
}
